tools/build-tools/src/typeValidator/testGeneration.ts

```diff
@@ -56,7 +56,7 @@ import * as current from "${indexPath}/index";
             testString.push(`* If breaking change required, add in package.json under typeValidation.broken.${oldProjectData.packageDetails.version}:`);
             testString.push(`* "${getFullTypeName(currentType)}": {"forwardCompat": false}`);
             testString.push("*/");
-            testString.push(...  buildTestCase(oldType, currentType, brokenData?.forwardCompat ?? true));
+            testString.push(...  buildTestCase(oldType, currentType, brokenData?.forwardCompat ?? true, false, currentTypeData === undefined));
```

Comment:
i think we should literally add a removed field to the type TestCaseTypeData, and set it when we create it above. only the current data type will be removed, the old type will never be removed

Reply:
after this, you will not need to modify the function signature, just check the remove field on the TestCaseTypeData's passing in